src/coreclr/tools/aot/ILCompiler.MetadataTransform/ILCompiler/Metadata/Transform.Method.cs

```diff
@@ -61,7 +61,6 @@ private Method HandleMethodDefinition(Cts.MethodDesc method)
         private void InitializeMethodDefinition(Cts.MethodDesc entity, Method record)
         {
             record.Name = HandleString(entity.Name);
-            record.Signature = HandleMethodSignature(entity.Signature);
```

Comment:
Dtto

Reply:
This is never reached for non-ecma methods. There is an else branch that throws NotImplemented. There's no scenario for non-ECMA metadata generation.

All of this should be eventually rewritten to work on top of S.R.Metadata instead of the type system.